.github/workflows/build-and-publish.yaml

```diff
@@ -22,3 +22,69 @@ jobs:
       ossrh-username: ${{ secrets.SONATYPE_OSSRH_USERNAME }}
       ossrh-password: ${{ secrets.SONATYPE_OSSRH_PASSWORD }}
       github-token: ${{ secrets.GH_TOKEN }}
+  docs:
```

Comment:
Maybe we should add this to our java-library workflow 🤔 then we can reuse it in other repos and we can restructure the workflow a bit 
<img width="2922" height="198" alt="image" src="https://github.com/user-attachments/assets/3926a392-25bb-452d-b27f-a07f452f99fc" />
I think it makes sense to have the "Build Docs" job run in parallel to the java build and then the publish docs job depends on assess code quality and "Build docs"
<img width="3033" height="490" alt="image" src="https://github.com/user-attachments/assets/fb7e7829-9c40-493f-8d15-0a5f93a8d9a1" />

Reply:
Ideally we have an input param build-docs which is false by default